plugins/bcollection/_menu.py

```diff
@@ -0,0 +1,311 @@
+# -*- coding: utf-8 -*-
+"""
+Copyright (c) 2019 Fernando Póvoa (sbrubes)
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+"""
+from time import sleep
+
+import xl
+import xlgui
+
+from gi.repository import GLib
+from gi.repository import Gio
+from gi.repository import Gtk
+
+from xl.common import threaded
+from xl.nls import gettext as _
+from xlgui.widgets import menu, dialogs
+
+from _base import get_name, Logger
+from _database import execute_delete
+from _utils import get_icon
+
+#: Logger: Logger to file
+_LOGGER = Logger(__name__)
+
+#: str
+_CONTEXT_MENU_NAME = get_name('tree-view-menu')
+
+
+@xl.common.threaded
+def read_tags(tracks):
+    """
+        Threaded read tags notifying changes
+        :param tracks: list
+        :return:  None
+    """
+    for i in tracks:
+        i.read_tags()
+        sleep(0.055)
```

Comment:
Without the sleep I'll not need to read tags threaded, since it freezes the playlist anyway.
So, it should stay, unless we remove this "feature".

Reply:
Don't you want to yield the thread then? Something like `os.sched_yield()`? There is no meaning for this magical constant `0.055`, right? This means that it will not be sufficient on slower machines and way too much on faster machines.

Reply:
I've done more tests and seems fine without the sleep, so I've removed it: https://github.com/exaile/exaile/pull/593/commits/e3d92e2198322f0c500952ff408307899e1f9790